runtime/v2/shim/shim.go

```diff
@@ -132,6 +135,7 @@ const (
 func parseFlags() {
 	flag.BoolVar(&debugFlag, "debug", false, "enable debug output in logs")
 	flag.BoolVar(&versionFlag, "v", false, "show the shim version and exit")
+	flag.BoolVar(&infoFlag, "info", false, "get the option protobuf from stdin, print the shim info protobuf to stdout, and exit")
```

Comment:
I have a small suggestion, `info` looks better as a subcommand, unlike `-v`, it will be called by runtime like `shim start` and `shim delete`, and the information returned by info will be used externally (ctr, grpc client).
we may not require shim to implement it like start and delete, but we should try to suggest shim to provide info commands as well

Reply:
The shim CLI is not for human users, so I don't see a strong reason to make it a subcommand.

Reply:
Turned out that this shouldn't be a subcommand.
Because old shims produce very confusing errors for unknown subcommands
```
$ containerd-shim-runc-v2 info
io.containerd.runc.v2: shim namespace cannot be empty
```

Reply:
Oh, it's frustrating :smiling_face_with_tear:

Reply:
For old shims, the unknown flag `-info` also outputs a lot of information.
```bash
$  containerd-shim-runc-v2 -info
flag provided but not defined: -info
Usage of containerd-shim-runc-v2:
  -address string
        grpc address back to main containerd
  -bundle string
        path to the bundle if not workdir
  -debug
        enable debug output in logs
  -id string
        id of the task
  -namespace string
        namespace that owns the shim
  -publish-binary string
        path to publish binary (used for publishing events), but containerd-shim-runc-v2 will ignore this flag, please use the TTRPC_ADDRESS env
  -socket string
        socket path to serve
  -v    show the shim version and exit
```

These error messages look a bit bloated when returned as error.

Also, I don't think we can strictly expect other old shims to always return errors that match our expectations when executing `shim -info`

Of course this is not a blocking issue

Reply:
> These error messages look a bit bloated when returned as error.

Yes, but "flag provided but not defined: -info" is still better than "io.containerd.runc.v2: shim namespace cannot be empty"